arraycontext/impl/pyopencl/fake_numpy.py

```diff
@@ -208,6 +210,31 @@ def all(self, a):
             result = result.get()[()]
         return result
 
+    def array_equal(self, a, b):
+        def as_scalar(bool_value):
+            import numpy as np
+            return self._array_context.from_numpy(
+                np.array(int(bool_value), dtype=np.int8))
+
+        if type(a) != type(b):
+            result = as_scalar(False)
+        elif not is_array_container(a):
+            same_shape = a.shape == b.shape
+            if same_shape:
+                result = (a == b).all()
+            else:
+                result = as_scalar(False)
+        else:
+            queue = self._array_context.queue
+            result = rec_multimap_reduce_array_container(
+                    partial(reduce, partial(cl_array.minimum, queue=queue)),
```

Comment:
Is `minimum` documented to do the right thing applied to `bool` arrays?

Reply:
I separated the device-to-host conversion from the recursion step, so I think this should only be operating on `int8` arrays now.